src/GUI/UI/UIHouses.h

```diff
@@ -40,6 +38,24 @@ bool houseDialogPressEscape();
  */
 void playHouseSound(HOUSE_ID houseID, HouseSoundType type);
 
+enum class HouseInteractiveType {
+    HOUSE_INTERACTIVE_PROPRIETOR,
+    HOUSE_INTERACTIVE_NPC,
+    HOUSE_INTERACTIVE_TRANSITION
+};
+using enum HouseInteractiveType;
+
+struct HouseInteractiveDesc {
+    HouseInteractiveType type;
+    std::string label = "";
+    GraphicsImage *icon = nullptr;
+    GUIButton *button = nullptr;
+    union {
+        int targetMapID;
+        NPCData *npc;
+    } data{};
```

Comment:
Let's refrain from adding more unsafe unions.

Either just add both fields or add a private `std::variant` & accessor methods that check `type`.